app/assets/javascripts/controllers/middleware_server/middleware_server_controller.js

```diff
@@ -0,0 +1,146 @@
+ManageIQ.angular.app.controller('mwServerController', MwServerController);
+
+MwServerController.$inject = ['$scope', 'miqService' ];
+
+/**
+ * MwServerController - since there can be only one controller per page due to:
+ * 'ManageIQ.angular.scope = $scope;' since this is needed by miqCallAngular to
+ * showListener via ManageIQ.angular.scope.$apply.
+ * This is the parent controller for the page that is bootstrapped,
+ * interacting with the page via $scope and then $broadcast events down to the sub
+ * controllers to handle them in isolation.
+ *
+ * Controller Hierarchy is:
+ * - MwServerController
+ * -- MwServerOpsController
+ * -- MwAddDeploymentController
+ * -- *Any other controllers (more coming...)
+ *
+ * This is certainly not ideal, but allows us to use multiple controllers on a page.
+ * And provides loose coupling of controllers via events instead of depending on
+ * parent/child controller relationships.
+ * @param $scope
+ * @param miqService
+ * @constructor
+ */
+function MwServerController($scope, miqService) {
+  ManageIQ.angular.scope = $scope;
+
+  /////////////////////////////////////////////////////////////////////////
+  // Server Ops
+  /////////////////////////////////////////////////////////////////////////
+
+  $scope.showServerOpsListener = function (args) {
+    var operation = args.split(':')[1]; // format is 'operation:resume'
+
+    $scope.paramsModel = {};
+    if (operation) {
+      $scope.paramsModel.serverId = angular.element('#mw_param_server_id').val();
+      $scope.paramsModel.operation = operation;
+      $scope.paramsModel.operationTitle = makeOperationDisplayName(operation) + ' ' + _('Server');
+      $scope.paramsModel.operationButtonName = makeOperationDisplayName(operation);
+    }
+    $scope.paramsModel.timeout = 10; // default timeout value
+  };
+
+  $scope.runOperation = function () {
+    $scope.$broadcast('mwSeverOpsEvent', $scope.paramsModel);
+  };
+
+  var makeOperationDisplayName = function (operation) {
+    return _.capitalize(operation);
+  };
+
+  /////////////////////////////////////////////////////////////////////////
+  // Add Deployment
+  /////////////////////////////////////////////////////////////////////////
+
+  $scope.deployAddModel = {};
+  $scope.deployAddModel.enableDeployment = true;
+  $scope.deployAddModel.serverId = angular.element('#server_id').val();
+
+  $scope.showDeployListener = function () {
+    $scope.deployAddModel.showDeployModal = true;
+    $scope.resetDeployForm();
+  };
+
+  $scope.resetDeployForm = function () {
+    $scope.deployAddModel.enableDeployment = true;
+    $scope.deployAddModel.runtimeName = undefined;
+    $scope.deployAddModel.filePath = undefined;
+    angular.element('#deploy_div :file#upload_file').val('');
+    angular.element('#deploy_div input[type="text"]:disabled').val('');
+  };
+
+  $scope.$watch('filePath', function(newValue) {
```

Comment:
This should be `$scope.$watch('deployAddModel.filePath', function(newValue) {`.
(the impact of it can be seen since runtime name is not automatically filled when selecting a file)